### PR TITLE
[ABW-3565] Show resource details on Asset Transfer

### PR DESCRIPTION
**Pull request by matiasbzurovski**
Jira ticket: [ABW-3565](https://radixdlt.atlassian.net/browse/ABW-3566)

## Description
Allows to show resource details on Asset Transfer flow.

## Video

https://github.com/user-attachments/assets/7c9e1dbd-52a2-4a0a-9254-bb984a6dca00

[ABW-3565]: https://radixdlt.atlassian.net/browse/ABW-3565?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by GhenadieVP — CHANGES_REQUESTED** [review]
When showing Stake Units, Stake claim NFTS and pool units, their respective details screen needs to be shown. Those should not be shown as Fungible or NonFungibleToken details.

See Android Demo for reference -> https://github.com/radixdlt/babylon-wallet-android/pull/1048

**Review by GhenadieVP — APPROVED** [review]
LGTM!
Very nice, some minor comments added.

File: RadixWallet/Features/AssetTransferFeature/Components/TransferAccountList/ReceivingAccount/Asset/NonFungibleResourceAsset+Reducer.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import ComposableArchitecture
+
+public struct NonFungibleResourceAsset: Sendable, FeatureReducer {
+	public struct State: Sendable, Hashable, Identifiable {
+		public typealias ID = String
+		public var id: ID { token.id.toRawString() }
```
[the remark this answers is not in the document]
Comment (matiasbzurovski) [reply RadixWallet/Features/AssetTransferFeature/Components/TransferAccountList/ReceivingAccount/Asset/NonFungibleResourceAsset+Reducer.swift:6]: Actually the `nftGlobalID` isn't used at all, so removing that var.

File: RadixWallet/Features/AssetTransferFeature/Components/TransferAccountList/ReceivingAccount/Asset/NonFungibleResourceAsset+Reducer.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	}
+
+	public init() {}
+
+	public func reduce(into state: inout State, viewAction: ViewAction) -> Effect<Action> {
+		switch viewAction {
+		case .resourceTapped:
+			.send(.delegate(.resourceTapped))
+		}
+	}
```
[the remark this answers is not in the document]
Comment (matiasbzurovski) [reply RadixWallet/Features/AssetTransferFeature/Components/TransferAccountList/ReceivingAccount/Asset/NonFungibleResourceAsset+Reducer.swift:33]: if we look at this view + reducer only, then the logical choice would be to remove it. However, given this is used in the context of a parent view (`ResourceAsset.View`) that uses `CaseLet` to define which child view to display (fungible vs non fungible), it adds even more noise if we make one of them a plain SwiftUI view.

**Pull request merged**